Quirks are now handled by the mind (and other quirk code changes)

tontyGH opened the pull request:
## About The Pull Request

* (!!) Quirks are now assigned and handled inside the mind
* (!!) When the mind transfers, quirks will "transfer" with it (they simply retarget to the new mob)
* Closes #6456
* Cloning/Blood data no longer get quirks
* Health analyzers use last_mind when checking quirks

Misc. quirk code changes:
* (!!) all previous instances of quirk_holder has been renamed to quirk_target (rip diffs)
* (!) human_only is replaced with a more specific restricted_mobtype variable for more flexibility (i.e it could affect you as any living mob or only as a human), however the functionality of quirks were not changed in this PR
* (!) added a restricted_species proc to specify what species this would affect on humans, with a boolean for whitelist/blacklist functionality
* (!) the phobia type has been turned into a functionally identical "trauma" type and paraplegic is now a subtype of it (no this won't break prefs)
* 2 procs were removed as they were unused/identical to another proc elsewhere
* other things that I'm probably forgetting

## Why It's Good For The Game

Quirks are meant to be personal characteristics, but they currently are handled in the mob's body, which causes anybody who's transferred/removed from that body to gain/lose those quirks. This leads to strange situations such as completely and permanently losing interest in your heirloom when you are monkeyized, and being put inside the body of a mute making you unable to speak, even if you yourself aren't mute.

Moving them to the mind makes them stay attached to their respective character, and also makes transferring quirks easier.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

demonstrates that quirks are preserved across mind transfers, previously this would have removed all of your quirks

https://user-images.githubusercontent.com/39193182/226461461-f1659560-0e29-4455-8bea-f282c81e1795.mp4

</details>

## Changelog
:cl: tonty
tweak: Quirks are no longer bound to the body, instead being assigned to the character itself
fix: Having your character transferred (ex: cloning, monkeyize) should no longer remove your quirks
/:cl:

tontyGH committed:
renames quirk_holder to quirk_target

tontyGH committed:
mind now handles quirks

moved /roundstart_quirks from living_defines.dm to mind.dm and renamed to /quirks

moves (most of) the status_procs.dm quirk procs to mind.dm
(if you have a suggestion for a better spot please say)

removed the transfer procs in _quirk.dm in favor of mind.dm's transfer_to

cleaned up _quirk.dm

tontyGH committed:
misc fixes/adjustments

tontyGH committed:
more quirk changes

EvilDragonfiend commented on code/datums/traits/_quirk.dm:
```diff
@@ -4,120 +4,135 @@
 	var/desc = "This is a test quirk."
 	var/value = 0
-	var/human_only = TRUE
+	var/mob/living/restricted_type = /mob/living/carbon/human //specifies which mob subtype can use this trait
```
I think this should be list type, not a single variable. If I say an example, there can be a quirk like "Overoxygionated" that can increase the sprite size of moth/apid/and flyperson characters. (as insects with a lot of oxygen can increase their size in real), but it will be nothing to non-insects.

There is an example of such case as `restricted_species = list(SPECIES_LIZARD)` in `uplink_items.dm`

tontyGH replied:
addressed in the newest commit

EvilDragonfiend commented on code/datums/traits/positive_quirk.dm:
```diff
@@ -123,10 +124,20 @@
+		quirk_target.grant_language(known_language, TRUE, TRUE, LANGUAGE_MULTILINGUAL)
 //Credit To Yowii/Yoworii/Yorii for a much more streamlined method of language library building
 
+/datum/quirk/multilingual/add()
```
I think this can possibly make a monkified person talk this language.

tontyGH replied:
you are a wizard for seeing this

tontyGH committed:
multilingual fix

how the fuck did he spot this bug

tontyGH committed:
better quirk restriction variables

i used /datum/species instead of the SPECIES defines for subtyping, so if you had a quirk that applied to lizards they would also apply to ashwalkers

PowerfulBacon commented on code/datums/traits/_quirk.dm:
```diff
@@ -4,120 +4,151 @@
+			REMOVE_TRAIT(quirk_target, mob_trait, ROUNDSTART_TRAIT)
+		remove()
+	quirk_target = to_mob
+	if(process)
+		START_PROCESSING(SSquirks, src)
```
This doesnt stop processing at any point, if you transfer a mob while keeping the original mob alive issues may arise

tontyGH replied:
This shouldn't be an issue since quirk_target is changed on transfer, and from what I've tested none of the processing quirks continue to apply onto the old mob so it should be fine

PowerfulBacon replied:
START_PROCESS has a flag check to prevent double processing, this is fine
